metastore/src/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java

```diff
@@ -2151,4 +2156,24 @@ public AggrStats getAggrColStatsFor(String dbName, String tblName,
     PartitionsStatsRequest req = new PartitionsStatsRequest(dbName, tblName, colNames, partNames);
     return client.get_aggr_stats_for(req);
   }
+
+  //multiple clients may initialize the hook at the same time
+  synchronized private URIResolverHook loadUriResolverHook() throws IllegalStateException {
+
+    String uriResolverClassName =
+            conf.get(ConfVars.METASTORE_URI_RESOLVER.toString(), "").trim();
```

Comment:
If METASTORE_URI_RESOLVER is not specified, will this effect in an empty string or null?

Reply:
the default is empty string (defined in the variable in HiveConf.java)